### ISSUE TITLE
kustomize build fails with remote transformer function with kustomize version 4.5.5+

### ISSUE THREAD

**Issue by Jell** [kubernetes-sigs/kustomize#4958]
### What happened?

Running:

```bash
kustomize build --enable-alpha-plugins .
```

with this kustomization file:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
transformers:
  - git@github.com:Jell/devops-test-tooling
```

returns:

```
Error: couldn't execute function: chdir /tmp/kustomize-2989599739: no such file or directory
```

when using `kustomize` version 4.5.5, 4.5.6 and 4.5.7, but works on all earlier 4.5.X versions

### What did you expect to happen?

```bash
kustomize build --enable-alpha-plugins .
```

should work the same with regards to remote transformers as with version `4.5.4`

### How can we reproduce it (as minimally and precisely as possible)?

Create a kustomization file that points to a remote transformer function

kustomization.yaml:
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
transformers:
  - git@github.com:Jell/devops-test-tooling
```

the URL points to a folder with those two files:

config.yaml
```yaml
apiVersion: examples.config.kubernetes.io/v1beta1
kind: cloudproxy
metadata:
  name: cloudproxy
  annotations:
    config.kubernetes.io/function: |-
      container:
        image: europe-west3-docker.pkg.dev/devops-test-2/devops-test/kustomize-transformer-cloud-sql-proxy:main
```

kustomization.yaml:
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - config.yaml
```

Checkout: https://github.com/Jell/kustomize-issue-4958 for reproducing the error.

### Expected output

same output as with kustomize version 4.5.4

### Actual output

```
Error: couldn't execute function: chdir /tmp/kustomize-2989599739: no such file or directory
```

### Kustomize version

4.5.7

### Operating system

Linux

**Comment by Jell** [comment]
Made a repo with a minimum setup to reproduce the error: https://github.com/Jell/kustomize-issue-4958/blob/main/README.md

**Comment by Jell** [comment]
the issue appears to be with remote SSH URLs, and both public and private SSH URLs are affected.

**Comment by KnVerey** [comment]
Thank you for the detailed reproduction! I can reproduce with v4.5.7, but I believe we've already merged a fix for this; kustomize built from current master attempts to pull the container. It looks like a problem we had related to working directories with KRM functions, and indeed checking out a previous revision confirms it was #4654 that fixed it. I'm going to close this since the fix is already in place. We are planning a release soon, hopefully next week.

/close
/triage resolved
